### DISCUSSION TITLE
How to run expressions that use JSON tags?

### DISCUSSION THREAD

**Discussion by star2478** [antonmedv/expr#731] [Q&A] [answered]
Hi
I have defined a struct that contains JSON tags. 
```go
type UserModel struct {
	UserID    string `json:"user_id"`
	Employee  int    `json:"employee"`
	Mobile    string `json:"mobile"`
	Email     string `json:"email"`
}
```
Expressions like ``UserID=="xxxx"?Employee:777`` can run, but ``user_id=="xxxx"?employee:777`` cannot.  How can I make ``user_id=="xxxx"?employee:777`` run as well?

**Comment by antonmedv** [comment, marked as answer]
You can use expr tags:

```go
type UserModel struct {
	UserID    string `expr:"user_id"`
	Employee  int    `expr:"employee"`
	Mobile    string `expr:"mobile"`
	Email     string `expr:"email"`
}

**Reply by star2478** [reply]
If UserModel is in a third-party library and I am unable to add an expr tag to it, what should I do?

**Reply by antonmedv** [reply]
I see. Probably we can add an option to allow using `json` tags.

**Reply by star2478** [reply]
That's great! When are we expected to be able to use this feature?

**Reply by antonmedv** [reply]
I don't have a specific timeline. But if you would support Expr development via GitHub Sponsor https://github.com/sponsors/antonmedv/sponsorships?sponsor=antonmedv&tier_id=315532&preview=false I will try to add this feature this week.

**Reply by star2478** [reply]
Another way is to convert struct to json and then json to map, which can also use keys in json tags.

```go
userModel := User{
	UserID:   "good",
	Employee: 1,
	Mobile:   "13800001111",
}
var user map[string]interface{}
jsonData, _ := json.Marshal(userModel)
json.Unmarshal(jsonData, &user)
exprAppend = `user_id=="good"?"11":"22"`
if value, err := expr.Eval(exprAppend, user); err != nil {
	panic(err)
} else {
	fmt.Println("result: ", value)
}
```
